arbeitszeit_flask/api/response_documentation.py

```diff
@@ -0,0 +1,23 @@
+from flask_restx import Namespace
+
+from arbeitszeit_flask.api.schema_converter import json_schema_to_flaskx
+from arbeitszeit_web.api_presenters.response_errors import ApiResponseError
+
+
+class with_response_documentation:
+    def __init__(
+        self, error_responses: list[type[ApiResponseError]], namespace: Namespace
+    ) -> None:
+        self._error_responses = error_responses
+        self._namespace = namespace
+
+    def __call__(self, original_function):
+        for response in self._error_responses:
+            error_schema = json_schema_to_flaskx(
+                schema=response.get_schema(), namespace=self._namespace
+            )
+            decorator = self._namespace.response(
+                code=response.code, description=response.description, model=error_schema
+            )
+            decorated_fn = decorator(original_function)
```

Comment:
This doesn't look right. I seems like only ever the last decorator constructed is actually applied to the original function.

Reply:
You are right, it works. The reason why this works is that `flask-restx` violates the convention on how decorators should work.

Usually a decorator accepts a function as an argument and returns a new function while leaving the old function unchanged. Although `Namespace.response` from `flask-restx` does return a new function it also modifies the original function.

See here for the implementation: https://github.com/python-restx/flask-restx/blob/master/flask_restx/namespace.py#L129

Since this is code outside of our project it is probably okay to leave it as is. I would still prefer

```python
decorated_fn = original_function # <---
for response in self._error_responses:
    ...
    decorated_fn = decorator(decorated_fn)  # <-- decorated_fn instead of original_function
return decorated_fn
```